### PR TITLE
Feature/vote to int

### LINKED ISSUE #149: `PlaintextBallotSelection` `plaintext` field should be an `int`
## Issue

This is the sibling issue for:

https://github.com/microsoft/electionguard-python/issues/34

**Description**

Since we added the extended_data field to capture text, the plaintext field can be replaced with an integer.

As a string, this filed is flexible and supports an extended range of use cases without having to modify the data structure, but at the expense of added parsing complexity. Since this field really is just a value that is encrypted, it could be an integer and imply to the consumer that this is an encodable value and that the extended_data field should be used for any string representations. One other consideration is that this decreases the human readability of a plaintext ballot expressed in json (unless we further add parsing complexity).

in such an implementation, for n-of-m type contests; the value 0 should always resolve to False whereas any other value should resolve to True

### PR DESCRIPTION
**Pull request by AddressXception**
🚨Please review the [guidelines for contributing](../CONTRIBUTING.md) to this repository.

### Issue
*Link your PR to an issue*

Fixes #149  and #141

### Description
change the data type of the PlaintextBallotSelection vote field to uint64

### Testing
1. `make test`
2. `make build-netstandard`
3. open visual studio for mac and run the binding project unit tests

### CODE REVIEW

**Comment by AddressXception** [comment]
this ended up a bit larger than i would have liked

File: bindings/netstandard/ElectionGuard/ElectionGuard.Shared/ElectionguardSafeHandle.cs [hunk 1]
```diff
@@ -1,18 +1,49 @@
...
         {
-            //MyMathFuncsWrapper.DisposeMyMathFuncs(handle);
-            return true;
+            try
+            {
+                return Free();
+            }
+            catch(Exception ex)
+            {
+                Console.WriteLine($"{nameof(T)} ERROR ReleaseHandle: {ex.Message}");
```
Comment (AddressXception) [inline bindings/netstandard/ElectionGuard/ElectionGuard.Shared/ElectionguardSafeHandle.cs:44]: just logging this out for now

File: bindings/netstandard/ElectionGuard/ElectionGuard.Shared/NativeInterface.cs [hunk 2]
```diff
@@ -233,18 +234,44 @@ internal static unsafe class PlaintextBallotSelection
         {
             internal unsafe struct PlaintextBallotSelectionType { };
 
+            internal class PlaintextBallotSelectionHandle
```
Comment (AddressXception) [inline bindings/netstandard/ElectionGuard/ElectionGuard.Shared/NativeInterface.cs:237]: only classes touched in this PR are updated to this new format in order to keep the changes scoped.

File: src/electionguard/random.hpp [hunk 3]
```diff
@@ -17,24 +17,60 @@ using namespace std;
...
+    /// </summary>
     class Random
     {
       public:
-        /// https://nvlpubs.nist.gov/nistpubs/SpecialPublications/NIST.SP.800-90Ar1.pdf
+        /// <summary>
+        /// Get pseudo-random bytes for the specified ByteSize.
+        ///
+        /// Note:
+        /// This implmenetation expects the OS random source to have at least `size * 2`
```
Comment (AddressXception) [inline src/electionguard/random.hpp:47]: i didnt want to get too specific with the comment, but this should not be an issue on any of our target hardware.  If, for instance someone tried to run this on an ultra low powered microcontroller its possible the entropy source may be exhausted depending on what other tasks the operating system is executing.  there's no good way for us to control for that, so consumers just need to know that dev/urandom guarantees a quick response, but not high quality bits.

File: src/electionguard/ballot.cpp [hunk 4]
```diff
@@ -59,8 +61,26 @@ namespace electionguard
...
 
-    uint64_t PlaintextBallotSelection::toInt() const { return stoul(pimpl->vote); }
     string PlaintextBallotSelection::getObjectId() const { return pimpl->object_id; }
+    uint64_t PlaintextBallotSelection::getVote() const { return pimpl->vote; }
+    bool PlaintextBallotSelection::getIsPlaceholder() const
+    {
+        return pimpl->isPlaceholderSelection;
+    }
+
+    bool PlaintextBallotSelection::isValid(const std::string &expectedObjectId) const
```
Comment (AddressXception) [inline src/electionguard/ballot.cpp:71]: important to note that the constructor allows a bad value in but the isValid check fails for any value other than 0 or 1 currently.  this is to enable better future support for other vote types without requiring a change to the public API

**Review by keithrfung — APPROVED** [review]
lgtm.

**Pull request merged**